modules/ref_device_module/src/ref_device_impl.cpp

```diff
@@ -123,7 +123,10 @@ void RefDeviceImpl::initClock()
 
     microSecondsFromEpochToDeviceStart = std::chrono::duration_cast<std::chrono::microseconds>(startAbsTime.time_since_epoch());
 
-    this->setDeviceDomain(DeviceDomain(RefChannelImpl::getResolution(), RefChannelImpl::getEpoch(), UnitBuilder().setName("second").setSymbol("s").setQuantity("time").build()));
+    this->setDeviceDomain(DeviceDomain(RefChannelImpl::getResolution(),
+                                       RefChannelImpl::getEpoch(),
+                                       UnitBuilder().setName("second").setSymbol("s").setQuantity("time").build(),
+                                       ReferenceDomainInfoBuilder().setReferenceDomainId(localId).setReferenceDomainOffset(0).build()));
```

Comment:
I'd set the reference domain ID to the serial number. Maybe "manufacturer_serial".